smalivm/src/test/java/org/cf/smalivm/TestMethodInheritence.java

```diff
@@ -15,10 +15,10 @@
     public static class ChildClass {
         private static final String CLASS_NAME = "Lchild_class;";
         private static final String EXPECTED_VALUE = "abstract?";
-        private VirtualMachine vm;
+        private static VirtualMachine vm;
 
-        @Before
-        public void setupVM() {
+        @BeforeClass
```

Comment:
@Before is used because it's not guaranteed that VirtualMachine won't maintain some sort of state. So you could end up having tests interfere with each other. Most of the slow down from running tests is from SmaliClassManager, and within that, mostly from looking up reflib.zip entries. That caching can probably be made static, so it doesn't happen every time a SmaliClassManager is instantiated (and VirtualMachine). It might even make more sense to have a constructor that doesn't cache framework classes.